### ISSUE TITLE
Error building. Useradd not found.

### ISSUE THREAD

**Issue by blaze-developer** [ading2210/shimboot#80]
When building an image with ``sudo ./build_complete.sh dedede release=unstable desktop=kde``

It goes through a long process, but ultimately ends in
```
SELINUX=disabled
/opt/setup_rootfs.sh: line 120: useradd: command not found
```

If you are reporting an issue with the build process, please run the scripts in debug mode by putting `DEBUG=1` before the build commmand, like `sudo DEBUG=1 ./build_complete.sh`.

**Build Device (please complete the following information):**
 - OS: Arch Linux
 - Neofetch out
bryce@obsidian 
distro: Arch Linux x86_64 
kernel: 6.10.7-arch1-1 
uptime: 14 hours, 37 mins 
pacman: 1279 (pacman) 
wm: Hyprland 
memory: 3963MiB / 15702MiB

**Comment by ading2210** [comment]
Can you run the build script with `DEBUG=1` enabled and post the log?

**Comment by OtterCodes101** [comment]
can you run `ls /usr/sbin/useradd` for me?

**Comment by blaze-developer** [comment]
https://pastebin.com/zrVkCUB2,

```bash
❯ ls /usr/sbin/useradd
/usr/sbin/useradd
```

**Comment by OtterCodes101** [comment]
This means /usr/sbin is not in your path. Try running `export PATH=$PATH:/usr/sbin`, then try rebuilding the shimboot image.

**Comment by blaze-developer** [comment]
Wait but useradd is a perfectly working command on my system-

**Comment by OtterCodes101** [comment]
it is, but the directory that contains useradd isn't added to the list of directories your shell looks for executables in/

**Comment by blaze-developer** [comment]
But it seems that is is because I can simple run ``useradd`` and its a working command

**Comment by OtterCodes101** [comment]
without root?

**Comment by blaze-developer** [comment]
with and without

**Comment by OtterCodes101** [comment]
weird

**Comment by blaze-developer** [comment]
```
~
❯ which useradd
/usr/bin/useradd

~
❯ sudo which useradd
/usr/bin/useradd
```

**Comment by blaze-developer** [comment]
sorry if im being a problem u dont need today >-<

**Comment by Inglan** [comment]
Just FYI, I am experiencing this as well (building debain unstable with gnome)

**Comment by blaze-developer** [comment]
What distro are you using to build?

**Comment by Raf128** [comment]
He is using Arch Linux

**Comment by Inglan** [comment]
> What distro are you using to build? Because my personal guess is that the building script is designed for debian-based systems and somehow doesn't work for arch based systems. But if you are using a debian based distro or not arch, I would be proven incorrect.

Same command you are using, but with gnome instead of kde

**Comment by ading2210** [comment]
I wasn't able to reproduce this with an ARM build. I'll try it again at home once I have access to another x86 system to test on.

**Comment by ading2210** [comment]
I cannot reproduce this bug on any of the devices I tested for the build. I used the exact same build command too.

**Comment by ading2210** [comment]
It might be a problem with the version of debootstrap that Arch provides. The error that occurs is happening inside the Debian chroot, not on the host Arch system, which suggests that debootstrap didn't set the chroot up correctly.

**Comment by blaze-developer** [comment]
> It might be a problem with the version of debootstrap that Arch provides. The error that occurs is happening inside the Debian chroot, not on the host Arch system, which suggests that debootstrap didn't set the chroot up correctly.

I would agree, however debootstrap on the arch repository is the same debootstrap from debian.org.

https://archlinux.org/packages/extra/any/debootstrap/

**Comment by OtterCodes101** [comment]
can reproduce after attempting to run shimboot on arch

**Comment by goober3853** [comment]
So many people... that don't read the damn readme.

**Comment by blaze-developer** [comment]
> So many people... that don't read the damn readme.

How is 3 people *so many* people? And this thread is what *pointed out* the issue to be added to the readme and investigated.

**Comment by ading2210** [comment]
I do want the build scripts to work across multiple distros, but since I've never used Arch before it would be difficult for me to debug this.

**Comment by ading2210** [comment]
This has been fixed in commit 3892016

**Comment by Parzival-rp2** [comment]
getting the same issue on arch, even after running commands from the commit
